APNS.js

```diff
@@ -0,0 +1,95 @@
+var Parse = require('parse/node').Parse;
+// TODO: apn does not support the new HTTP/2 protocal. It is fine to use it in V1,
+// but probably we will replace it in the future.
+var apn = require('apn');
+
+/**
+ * Create a new connection to the APN service.
+ * @constructor
+ * @param {Object} args Arguments to config APNS connection
+ * @param {String} args.cert The filename of the connection certificate to load from disk, default is cert.pem
+ * @param {String} args.key The filename of the connection key to load from disk, default is key.pem
+ * @param {String} args.passphrase The passphrase for the connection key, if required
+ * @param {Boolean} args.production Specifies which environment to connect to: Production (if true) or Sandbox
+ */
+function APNS(args) {
+  this.sender = new apn.connection(args);
+
+  this.sender.on('connected', function() {
+      console.log('APNS Connected');
+  });
+
+  this.sender.on('transmissionError', function(errCode, notification, device) {
+    console.error('APNS Notification caused error: ' + errCode + ' for device ', device, notification);
+    // TODO: For error caseud by invalid deviceToken, we should mark those installations.
+  });
+
+  this.sender.on("timeout", function () {
+      console.log("APNS Connection Timeout");
+  });
+
+  this.sender.on("disconnected", function() {
+      console.log("APNS Disconnected");
+  });
+
+  this.sender.on("socketError", console.error);
+}
+
+/**
+ * Send apns request.
+ * @param {Object} data The data we need to send, the format is the same with api request body
+ * @param {Array} deviceTokens A array of device tokens
+ * @returns {Object} A promise which is resolved immediately
+ */
+APNS.prototype.send = function(data, deviceTokens) {
```

Comment:
No, internally the library just sends the notifications one by one, so we do not need to have size limitation.